### PR TITLE
Use prismarine-chunk for block entities 

### PR DESCRIPTION
**Pull request by extremeheat**
should be moved to prismarine-world soon, and prismarine-block should instead have the abstractions related to signs

### CODE REVIEW

File: lib/plugins/blocks.js [hunk 1]
```diff
@@ -583,7 +593,14 @@ function inject (bot, { version, storageBuilder }) {
   bot.canSeeBlock = canSeeBlock
   bot.blockAt = blockAt
   bot._updateBlockState = updateBlockState
-  bot._blockEntities = blockEntities
+  Object.defineProperty(bot, '_blockEntities', {
```
Comment (rom1504) [inline lib/plugins/blocks.js:596]: you can just remove that

File: lib/plugins/blocks.js [hunk 2]
```diff
@@ -451,19 +447,33 @@ function inject (bot, { version, storageBuilder }) {
...
-      prepareString(4)
-    ].join('\n')
+    const blockAt = column.getBlock(pos)
+    if (!blockAt.entity) {
+      blockAt.entity = nbt.comp({
+        id: nbt.string('minecraft:sign')
+      })
+    }
+
+    blockAt.entity.value.Text1 = nbt.string(prepareString(1))
```
Comment (rom1504) [inline lib/plugins/blocks.js:457]: can we put that in pchunk/pworld maybe ?

File: lib/plugins/blocks.js [hunk 3]
```diff
@@ -293,9 +283,6 @@ function inject (bot, { version, storageBuilder }) {
     }
     if (oldBlock.type !== newBlock.type) {
       const pos = point.floored()
-      delete blockEntities[pos]
-      delete signs[pos]
-
       const painting = paintingsByPos[pos]
```
Comment (rom1504) [inline lib/plugins/blocks.js:286]: we can probably make paintings not be in blockAt

File: lib/plugins/blocks.js [hunk 4]
```diff
@@ -254,9 +238,15 @@ function inject (bot, { version, storageBuilder }) {
...
 
     if (extraInfos) {
-      block.signText = signs[block.position]
       block.painting = paintingsByPos[block.position]
-      block.blockEntity = blockEntities[block.position]
+      if (block.entity) {
+        const id = block.entity.value.id.value
+        // Mege in abstractions related to signs
+        Object.assign(block.blockEntity, addExtraBlockEntityData(id, block.entity))
+        if (id === 'minecraft:sign' || id === 'Sign') {
```
Comment (rom1504) [inline lib/plugins/blocks.js:246]: this should be in pchunk/pworld

this is basically a more high level way to represent sign and there could be something like that for other block entities in the future

**Comment by rom1504** [comment]
https://github.com/PrismarineJS/mineflayer/pull/2441 this may help

**Comment by extremeheat** [comment]
Yeah, the changes to blocks.js in #2441 would fix the issue here with quotation marks. Not sure what's going on with the examples part of the PR

**Comment by rom1504** [comment]
can you maybe just pick the changes you like from that PR and put them here so we can merge this one?

File: lib/plugins/blocks.js [hunk 5]
```diff
@@ -335,11 +288,13 @@ function inject (bot, { version, storageBuilder }) {
     })
 
     if (typeof packet.blockEntities !== 'undefined') {
+      const column = bot.world.getColumn(packet.x, packet.z)
```
Comment (rom1504) [inline lib/plugins/blocks.js:291]: why not use chunk.loadBlockEntities ?

**Pull request merged**